ext/tests/post_hook_return_ignored_without_type.phpt

```diff
@@ -0,0 +1,16 @@
+--TEST--
+Check if post hook return value is ignored if return typehint not supplied
+--EXTENSIONS--
+opentelemetry
+--FILE--
+<?php
+\OpenTelemetry\Instrumentation\hook(null, 'helloWorld', post: fn(mixed $object, array $params, string $return) => 'ignored');
+
+function helloWorld(string $val) {
```

Comment:
As I mentioned on https://github.com/open-telemetry/opentelemetry-php-instrumentation/pull/79 it would be good to report somehow type violations otherwise it might be surprising and lead to some user confusions. It's open question how can we do that.

Reply:
I agree. We could just remove the restriction, but it does add an element of type safety. I only added this test as a form of documentation of the current behaviour - I didn't know the restriction was there.

Reply:
I don't have a solution yet, so I'll just add a TODO with some thoughts for now. Maybe we create an issue to track it?